File: Examples/CallGraph/Java/OrderService.java
/** Places an order by charging payment and persisting it. */
class OrderService {
    private final PaymentService payment;
    private final OrderRepository repository;

    OrderService(PaymentService payment, OrderRepository repository) {
        this.payment = payment;
        this.repository = repository;
    }

    void place(Order order) {
        payment.charge(order);
        repository.save(order);
    }
}

File: Examples/CallGraphDiff/TypeScript/Before/OrderService.ts
import { Order } from "./Order";
import { OrderRepository } from "./OrderRepository";
import { PaymentService } from "./PaymentService";

/** Places an order by charging payment and persisting it. */
export class OrderService {
    constructor(
        private readonly payment: PaymentService,
        private readonly repository: OrderRepository
    ) {}

    place(order: Order): void {
        this.payment.charge(order);
    }
}
